### PR TITLE
Follow up work for #1133: remove publishing-requirements.txt 

### PR DESCRIPTION
**Pull request by m1yag1**

### CODE REVIEW

File: roles/publishing_common/tasks/main.yml [hunk 1]
```diff
@@ -57,6 +65,15 @@
     virtualenv_python: python2
     state: latest
 
+- name: create the publishing virtualenv - new virtualenv in python2 folder
+  become: yes
+  become_user: "{{ venvs_owner|default('www-data') }}"
+  pip:
+    name: pip
+    virtualenv: "/var/cnx/venvs/python2/archive"
```
Comment (rnathuji) [inline roles/publishing_common/tasks/main.yml:73]: Should this be `/var/cnx/venvs/python2/publishing`?

**Comment by rnathuji** [comment]
Just FYI: when trying these changes locally really quick, I observed the following error in the first pass:

```
RUNNING HANDLER [publishing_common : add publishing] ***************************
task path: /vagrant/roles/publishing_common/handlers/main.yml:7
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/web_infrastructure/supervisorctl.py
Pipelining is enabled.
<local.cnx.org> ESTABLISH SSH CONNECTION FOR USER: None
<local.cnx.org> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/vagrant/.ansible/cp/d4c8dc3e82 local.cnx.org '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-xaerttrlvnppbcwfvrybwcwvoaqqcbql ; /usr/bin/python'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<local.cnx.org> (1, '\n{"msg": "ERROR: shutting down\\n", "failed": true, "state": "present", "name": "publishing", "invocation": {"module_args": {"username": null, "supervisorctl_path": null, "name": "publishing:", "signal": null, "server_url": null, "state": "present", "password": null, "config": null}}}\n', '')
<local.cnx.org> Failed to connect to the host via ssh: 
fatal: [local.cnx.org]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "config": null, 
            "name": "publishing:", 
            "password": null, 
            "server_url": null, 
            "signal": null, 
            "state": "present", 
            "supervisorctl_path": null, 
            "username": null
        }
    }, 
    "msg": "ERROR: shutting down\n", 
    "name": "publishing", 
    "state": "present"
}
```

Running again, it worked fine. Maybe there's some ordering issue? 

I can also run a clean build to see what happens.

**Comment by rnathuji** [comment]
An attempt from a clean build resulted in the following error in the next task:

```
RUNNING HANDLER [publishing_common : add publishing worker] ********************
task path: /vagrant/roles/publishing_common/handlers/main.yml:13
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/web_infrastructure/supervisorctl.py
Pipelining is enabled.
<local.cnx.org> ESTABLISH SSH CONNECTION FOR USER: None
<local.cnx.org> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/vagrant/.ansible/cp/d4c8dc3e82 local.cnx.org '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-lctyvpbjmaoikomfljcsoximilmirzcu ; /usr/bin/python'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<local.cnx.org> (1, '\n{"msg": "ERROR: no such process/group: publishing_worker\\n", "failed": true, "state": "present", "name": "publishing_worker", "invocation": {"module_args": {"username": null, "supervisorctl_path": null, "name": "publishing_worker:", "signal": null, "server_url": null, "state": "present", "password": null, "config": null}}}\n', '')
<local.cnx.org> Failed to connect to the host via ssh: 
fatal: [local.cnx.org]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "config": null, 
            "name": "publishing_worker:", 
            "password": null, 
            "server_url": null, 
            "signal": null, 
            "state": "present", 
            "supervisorctl_path": null, 
            "username": null
        }
    }, 
    "msg": "ERROR: no such process/group: publishing_worker\n", 
    "name": "publishing_worker", 
    "state": "present"
}
```

**Comment by m1yag1** [comment]
Weird, it completed for me just fine. I'll try to reproduce on my end.

**Comment by rnathuji** [comment]
Doing a retry after the first failure on a clean build, the publishing related issue seemed to be resolve itself (e.g. it's a "second pass works" issue), but I hit [this error](https://github.com/openstax/cnx/issues/1256) which is unrelated as it reproduces on `master` currently independent of these changes.

**Comment by m1yag1** [comment]
I think the problem here is I added both to the same place when the publishing_worker: is setup in another task file. The first pass fails b/c supervisor has yet to be restarted so it doesn't know what `publishing_worker:` is but then the handlers all fire. The second pass works because the handlers have went off by this point and supervisor is restarted and sees `publshing_worker:`. That's my theory.

File: roles/publishing_common/tasks/main.yml (hunk 1 above)
Comment (m1yag1) [reply roles/publishing_common/tasks/main.yml:73]: omg

Comment (m1yag1) [reply roles/publishing_common/tasks/main.yml]: did i mention i did all this in 1.5 hours on Friday using mostly find and replace? ;) Thank goodness for code review. 🙏

**Comment by m1yag1** [comment]
Updating this with status. I've made two changes and ran from a pre-existing deployment and everything finished successfully. I'm currently doing another build and will update with how that went tomorrow.

**Comment by m1yag1** [comment]
My build failed w/ the libs.sh error. My assumption was we didn't need the ssh step b/c of the new dist.rhaptos.org site. I'm re-running to see if it decides to work.

**Comment by m1yag1** [comment]
this is the error in my buidout output

```
...
Installing 'Products.RhaptosSite', 'Pillow', 'demjson', 'Products.PloneHotfix20110531', 'Products.PloneHotfix20110720', 'Products.PloneHotfix20121106', 'Products.PloneHotfix20130618==1.1', 'Products.Zope_Hotfix_CVE_2010_3198', 'rhaptos.mathjax', 'rhaptos.atompub.plone', 'rhaptos.swordservice.plone', 'egenix_mx_base', 'demjson', 'elementtree', 'Products.CacheSetup', 'Products.CNXCaching', 'Products.CNXPloneSite', 'Products.CMFDiffTool', 'Products.PloneHotfix20110720', 'Products.PloneHotfix20110531'.
We have no distributions for Products.RhaptosSite that satisfies 'Products.RhaptosSite==1.49'.
Download error: (110, 'Connection timed out') -- Some packages may not be found!
Download error: BadStatusLine:  -- Some packages may not be found!
Download error: (-2, 'Name or service not known') -- Some packages may not be found!
Couldn't find index page for 'Products.RhaptosSite' (maybe misspelled?)
Getting distribution for 'Products.RhaptosSite==1.49'.
While:
  Installing instance.
  Getting distribution for 'Products.RhaptosSite==1.49'.
Error: Couldn't find a distribution for 'Products.RhaptosSite==1.49'.
```

**Comment by rnathuji** [comment]
I wonder if the pypi server at [`dist.cnx.org`](https://dist.cnx.org/) crashed during your install and might explain the download errors. It seems to be down now at least. I'll reopen [cnx 1256](https://github.com/openstax/cnx/issues/1256) accordingly.

**Comment by m1yag1** [comment]
@rnathuji we have success! I think the issue was the buildout-fail left some files unpopulated and that my next attempt "tricked" the playbook into thinking that it was fully installed but was not. When I destroyed my vagrant env and re-provisioned everything worked.

**Review by rnathuji — APPROVED** [review]

**Comment by rnathuji** [comment]
Oh, @m1yag1 - I just noticed the CI error where the travis scripts still refer to the `archive-requirements.txt` file.

**Comment by m1yag1** [comment]
Ye,a i wasn't sure if i should remove travis before or after this PR goes in.

**Comment by rnathuji** [comment]
Cool -- just wanted to point it out.

**Comment by m1yag1** [comment]
@rnathuji, I removed publishing-requirements and archive-requirements. I left press in there... not sure if that's super useful though. I don't expect t here to be conflicts inside of the press virtualenv so not sure how useful that test is.

**Pull request merged**